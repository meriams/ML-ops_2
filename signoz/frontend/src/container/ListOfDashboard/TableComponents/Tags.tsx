/* eslint-disable react/destructuring-assignment */
import { Tag } from 'antd';

import { Data } from '../index';

function Tags(data: Data['tags']): JSX.Element {
	return (
		<>
			{data.map((e) => (
				<Tag key={e}>{e}</Tag>
			))}
		</>
	);
}

export default Tags;
