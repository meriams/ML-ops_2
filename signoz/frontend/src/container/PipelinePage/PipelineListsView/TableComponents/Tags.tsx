import { Tag } from 'antd';

function Tags({ tags }: TagsProps): JSX.Element {
	return (
		<span>
			{tags?.map((tag) => (
				<Tag color="magenta" key={tag}>
					{tag}
				</Tag>
			))}
		</span>
	);
}

interface TagsProps {
	tags: Array<string>;
}

export default Tags;
