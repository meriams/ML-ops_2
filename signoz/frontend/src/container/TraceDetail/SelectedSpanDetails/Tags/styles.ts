import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	gap: 0.5rem;
	flex-direction: column;
	width: 100%;
`;
