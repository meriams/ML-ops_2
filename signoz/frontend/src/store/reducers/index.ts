import { combineReducers } from 'redux';

import appReducer from './app';
import globalTimeReducer from './global';
import { LogsReducer } from './logs';
import metricsReducers from './metric';
import { ServiceMapReducer } from './serviceMap';
import traceReducer from './trace';
import { usageDataReducer } from './usage';

const reducers = combineReducers({
	traces: traceReducer,
	usageDate: usageDataReducer,
	globalTime: globalTimeReducer,
	serviceMap: ServiceMapReducer,
	app: appReducer,
	metrics: metricsReducers,
	logs: LogsReducer,
});

export type AppState = ReturnType<typeof reducers>;

export default reducers;
