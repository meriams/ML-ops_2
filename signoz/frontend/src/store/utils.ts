import getLocalStorageKey from 'api/browser/localstorage/get';
import { LOCALSTORAGE } from 'constants/localStorage';
import { User } from 'types/reducer/app';

export const getInitialUserTokenRefreshToken = (): AuthTokenProps | null => {
	const accessJwt = getLocalStorageKey(LOCALSTORAGE.AUTH_TOKEN);
	const refreshJwt = getLocalStorageKey(LOCALSTORAGE.REFRESH_AUTH_TOKEN);

	if (accessJwt && refreshJwt) {
		return {
			accessJwt,
			refreshJwt,
		};
	}

	return null;
};

interface AuthTokenProps {
	accessJwt: User['accessJwt'];
	refreshJwt: User['refreshJwt'];
}
