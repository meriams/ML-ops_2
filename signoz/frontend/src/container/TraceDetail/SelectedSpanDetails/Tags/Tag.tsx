import { Tooltip } from 'antd';
import { useIsDarkMode } from 'hooks/useDarkMode';
import { Fragment, useMemo } from 'react';
import { ITraceTag } from 'types/api/trace/getTraceItem';

import EllipsedButton from '../EllipsedButton';
import { CustomSubText, CustomSubTitle, SubTextContainer } from '../styles';
import { CommonTagsProps } from '.';
import { Container } from './styles';

function Tag({ tags, onToggleHandler, setText }: TagProps): JSX.Element {
	const isDarkMode = useIsDarkMode();

	const { value, isEllipsed } = useMemo(() => {
		const value = tags.key === 'error' ? 'true' : tags.value;

		return {
			value,
			isEllipsed: value.length > 24,
		};
	}, [tags]);

	return (
		<Fragment key={JSON.stringify(tags)}>
			{tags.value && (
				<Container>
					<CustomSubTitle>{tags.key}</CustomSubTitle>
					<SubTextContainer isDarkMode={isDarkMode}>
						<Tooltip overlay={(): string => value}>
							<CustomSubText
								ellipsis={{
									rows: isEllipsed ? 2 : 0,
								}}
								isDarkMode={isDarkMode}
							>
								{value}
							</CustomSubText>

							{isEllipsed && (
								<EllipsedButton
									{...{
										event: tags.key,
										onToggleHandler,
										setText,
										value,
										buttonText: 'View full value',
									}}
								/>
							)}
						</Tooltip>
					</SubTextContainer>
				</Container>
			)}
		</Fragment>
	);
}

interface TagProps extends CommonTagsProps {
	tags: ITraceTag;
}

export default Tag;
