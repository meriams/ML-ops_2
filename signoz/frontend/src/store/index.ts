import {
	applyMiddleware,
	compose,
	legacy_createStore as createStore,
} from 'redux';
import thunk, { ThunkMiddleware } from 'redux-thunk';
import AppActions from 'types/actions';

import reducers, { AppState } from './reducers';

const composeEnhancers =
	(window && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const store = createStore(
	reducers,
	composeEnhancers(
		applyMiddleware(thunk as ThunkMiddleware<AppState, AppActions>),
	),
);

export type AppDispatch = typeof store.dispatch;

if (window !== undefined) {
	window.store = store;
}

export default store;
