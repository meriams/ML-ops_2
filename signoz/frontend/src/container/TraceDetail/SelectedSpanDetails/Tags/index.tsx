import { Input, List, Typography } from 'antd';
import ROUTES from 'constants/routes';
import { formUrlParams } from 'container/TraceDetail/utils';
import {
	ChangeEvent,
	Dispatch,
	SetStateAction,
	useCallback,
	useMemo,
	useState,
} from 'react';
import { useTranslation } from 'react-i18next';
import { ITraceTag } from 'types/api/trace/getTraceItem';

import { ModalText } from '..';
import { Container } from './styles';
import Tag from './Tag';

function Tags({
	tags,
	linkedSpans,
	onToggleHandler,
	setText,
}: TagsProps): JSX.Element {
	const { t } = useTranslation(['traceDetails']);
	const [searchText, setSearchText] = useState('');
	const isSearchVisible = useMemo(() => tags.length > 5, [tags]);

	const getLink = useCallback(
		(item: Record<string, string>) =>
			`${ROUTES.TRACE}/${item.TraceId}${formUrlParams({
				spanId: item.SpanId,
				levelUp: 0,
				levelDown: 0,
			})}`,
		[],
	);

	const onChangeHandler = (e: ChangeEvent<HTMLInputElement>): void => {
		const { value } = e.target;
		setSearchText(value);
	};

	const filteredTags = tags.filter((tag) => tag.key.includes(searchText));

	if (tags.length === 0) {
		return <Typography>No tags in selected span</Typography>;
	}

	return (
		<Container>
			{isSearchVisible && (
				<Input.Search
					placeholder={t('traceDetails:search_tags')}
					allowClear
					onChange={onChangeHandler}
					value={searchText}
				/>
			)}
			{filteredTags.map((tag) => (
				<Tag
					key={JSON.stringify(tag)}
					{...{
						onToggleHandler,
						setText,
						tags: tag,
					}}
				/>
			))}
			{linkedSpans && linkedSpans.length > 0 && (
				<List
					header={<Typography.Title level={5}>Linked Spans</Typography.Title>}
					dataSource={linkedSpans}
					renderItem={(item): JSX.Element => (
						<List.Item>
							<Typography.Link href={getLink(item)}>{item.SpanId}</Typography.Link>
						</List.Item>
					)}
				/>
			)}
		</Container>
	);
}

interface TagsProps extends CommonTagsProps {
	tags: ITraceTag[];
	linkedSpans?: Record<string, string>[];
}

export interface CommonTagsProps {
	onToggleHandler: (state: boolean) => void;
	setText: Dispatch<SetStateAction<ModalText>>;
}

Tags.defaultProps = {
	linkedSpans: [],
};

export default Tags;
