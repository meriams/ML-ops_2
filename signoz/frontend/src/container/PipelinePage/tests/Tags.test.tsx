import { render } from '@testing-library/react';
import Tags from 'container/PipelinePage/PipelineListsView/TableComponents/Tags';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

const tags = ['server', 'app'];

describe('PipelinePage container test', () => {
	it('should render Tags section', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<Tags tags={tags} />
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
