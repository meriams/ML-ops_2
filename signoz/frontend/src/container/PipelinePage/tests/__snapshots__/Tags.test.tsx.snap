// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipelinePage container test should render Tags section 1`] = `
<DocumentFragment>
  <span>
    <span
      class="ant-tag ant-tag-magenta css-dev-only-do-not-override-2i2tap"
    >
      server
    </span>
    <span
      class="ant-tag ant-tag-magenta css-dev-only-do-not-override-2i2tap"
    >
      app
    </span>
  </span>
</DocumentFragment>
`;
